### PR TITLE
Refactor prov management

### LINKED ISSUE #26: Private and public session handling
- [x] separate own (autosaved) sessions from public sessions that all users can see
- [x] add public/private button for own sessions
- [x] add save button for private sessions

### PR DESCRIPTION
**Pull request by sgratzl**
fixes #26 #27 #28 #37 #63 #68 #75 #80

### CODE REVIEW

File: src/StartMenu.ts [hunk 1]
```diff
@@ -380,64 +375,56 @@ export class AEntryPointList implements IEntryPointList {
...
-            dialog.onSubmit(async () => {
-              const name = (<HTMLInputElement>document.getElementById('namedset_name')).value;
-              const description = (<HTMLInputElement>document.getElementById('namedset_description')).value;
-
+            editDialog(namedSet, async (name, description, isPublic) => {
               const params = {
                 name,
-                description
+                description,
+                permissions: isPublic? ALL_READ_READ: ALL_NONE_NONE
```
Comment (lehnerchristian) [inline src/StartMenu.ts]: missing space before colon

File: ordino/storage.py [hunk 2]
```diff
@@ -20,50 +20,63 @@ def get_namedsets():
...
+    del result['_id']
     return jsonify(result)
 
 
 def get_namedset_by_id(namedset_id):
   db = MongoClient(c.host, c.port)[c.database]
   q = dict(id=namedset_id)
   result = list(db.namedsets.find(q, {'_id': 0}))
-  if len(result) == 0:
+  if len(result) == 0 or not security.can_read(result[0]):
```
Comment (sgratzl) [inline ordino/storage.py]: this is not an rest method but an internal one

File: src/SessionList.ts [hunk 3]
```diff
@@ -2,111 +2,256 @@
...
+        </ul>
+        <div class="tab-content">
+            <div id="session_mine" class="tab-pane active">
+                <table class="table table-striped table-hover table-bordered table-condensed">
+                <thead>
+                  <tr>
+                    <th>Name</th>
+                    <th></th>
+                    <th>Date</th>
+                    <th></th>
```
Comment (lehnerchristian) [inline src/SessionList.ts]: We could add the headers "Visibility" and "Actions" (or other names) to the second and fourth <th> elements respectively

File: src/SessionList.ts [hunk 4]
```diff
@@ -2,111 +2,256 @@
...
+              </table>
+            </div>
+            <div id="session_others" class="tab-pane">
+                <table class="table table-striped table-hover table-bordered table-condensed">
+                <thead>
+                  <tr>
+                    <th>Name</th>
+                    <th>Creator</th>
+                    <th>Date</th>
+                    <th></th>
```
Comment (lehnerchristian) [inline src/SessionList.ts]: I think that here an "Actions" header instead of an empty one might also be good

File: src/SessionList.ts (hunk 3 above)
Comment (sgratzl) [inline src/SessionList.ts]: Visibility is quite a long word for showing just a single icon

File: src/StartMenu.ts [hunk 5]
```diff
@@ -335,40 +343,27 @@ export class AEntryPointList implements IEntryPointList {
     const that = this;
 
     const predefinedNamedSets = data.filter((d) => d.type !== ENamedSetType.NAMEDSET);
-    const customNamedSets = data.filter((d) => d.type === ENamedSetType.NAMEDSET);
+    const me = currentUserNameOrAnonymous();
+    const customNamedSets = data.filter((d) => d.type === ENamedSetType.NAMEDSET && d.creator === me);
+    const otherNamedSets = data.filter((d) => d.type === ENamedSetType.NAMEDSET && d.creator !== me);
```
Comment (lehnerchristian) [inline src/StartMenu.ts]: What if I create a named set and set it to public: should it be shown in "My Named Sets", in "Public Named Sets" or in both?

Comment (sgratzl) [inline src/StartMenu.ts]: atm it appears in "My"

File: src/SessionList.ts (hunk 3 above)
Comment (lehnerchristian) [inline src/SessionList.ts]: maybe "Access"? or would you prefer no header at all?

Comment (sgratzl) [inline src/SessionList.ts]: @mstreit any preference? same for the `Actions` header

Comment (mstreit) [inline src/SessionList.ts]: Access and Actions seem to be fine.

**Pull request merged**